ggshield/utils/files.py

```diff
@@ -32,6 +37,7 @@ def get_filepaths(
     recursive: bool,
     ignore_git: bool,
     ignore_git_staged: bool = False,
+    include_git_unstaged: bool = False,
```

Comment:
We should avoid introducing a forth boolean argument: 3 was already too many.

I suggest we introduce a FileListMode enum to replace them. It would have the following possible values:

| Value                   | recursive | ignore_git | ignore_git_staged | include_git_unstaged |
| ----------------------- | --------- | ---------- | ----------------- | -------------------- |
| FILES_ONLY              |           |            |                   |                      |
| GIT_COMMITTED           | x         |            | x                 |                      |
| GIT_COMMITTED_OR_STAGED | x         |            |                   |                      |
| ALL_BUT_GITIGNORED      | x         |            |                   | x                    |
| ALL                     | x         | x          |                   |                      |

The values do not include all possible flag combinations, but I think the others do not make sense. What do you think?

`GIT_COMMITTED_OR_STAGED` is a bit long, shorter suggestions are welcome :)